metadata_tools.py

```diff
@@ -39,173 +29,126 @@ def is_file_larger_than(self, filepath: str, size_in_mb) -> bool:
+
+    def read_iptc_metadata(self, path: str):
+        """reads iptc metadata of image and returns the dictionary"""
+        info = IPTCInfo(path, force=True)
+        return info
+
+    def attach_exif_metadata(self, path, exif_dict):
+        """attaches exif metadata tags to image using ExifTools subprocess in command line
+        args:
+            path: path to image
+            exif_dict: dictionary of exif terms using exif codes, and new values to assign"""
+
+        for exif_code, exif_value in exif_dict.items():
+            exif_code = self.exif_code_to_tag(exif_code)
+            command = ['exiftool', f"-{exif_code}={exif_value}", path]
+            subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        print("exif added succesfully")
+
+
+    # still doesn't work yet, but close
+    def remove_exif_tags(self, path, exif_codes):
+        """
+        Remove specified EXIF tags from an image.
+        args:
+            path: The path to the image.
+            exif_codes: A list of EXIF codes to remove.
+        """
+        try:
+            # Load EXIF
+            exif_data = self.read_exif_metadata(path=path, convert_tags=False)
+            # Iterate over the tags to remove
+            for code in exif_codes:
+                if code in exif_data:
+                    exif_data[code] = None
+
+            self.attach_exif_metadata(path=path, exif_dict=exif_data)
+
+            print("EXIF tags removed successfully")
+        except Exception as e:
+            print(f"Error removing EXIF tags: {e}")
+
+    def exif_code_to_tag(self, exif_code):
+        """converts exif code into the string of the tag name
+            args:
+                exif_code: the integer code of an exif tag to convert to TAG"""
+        tag_name = exifread.tags.EXIF_TAGS.get(exif_code, "Unknown Tag")
+        return tag_name[0]
+
+    def read_exif_metadata(self, path, convert_tags=True):
+        """reads and returns exif metadata, reads exif codes into TAG names
+            args:
+                path: path to image
+                convert_tags: True to convert tags to string, False keep exif codes
+        """
+        img = Image.open(path)
+        if convert_tags is True:
+            exif = {
+                PIL.ExifTags.TAGS[k]: v
+                for k, v in img._getexif().items()
+                if k in PIL.ExifTags.TAGS
+            }
+        else:
+            exif = img._getexif()
 
-if __name__ == "__main__":
-    print("Running tests...")
-    # md = MetadataTools('/Users/joe/web-asset-server/image_client/iptc_exif_test.jpg')
-    md = MetadataTools('test_image.jpg')
-    md.test_write_img_a()
+        img.close()
+        return exif
 
-    md.test_print_all_iptc()
 
-    # print(md)
+if __name__ == "__main__":
+    # print("Running tests...")
+    #
+    md = MetadataTools()
+
+    # # test code commented out
+    #
+    # md.iptc_attach_metadata(path='tests/test_images/test_image.jpg', iptc_dict={
```

Comment:
I probably should remove this test run code, not sure if I want to leave it around. maybe redundant due to unittests already existing